picard/ui/options/dialog.py

```diff
@@ -149,9 +149,13 @@ class OptionsDialog(PicardDialog, SingletonDialog):
     suspend_signals = False
 
     def add_pages(self, parent, default_page, parent_item):
-        pages = [(p.SORT_ORDER, p.NAME, p) for p in self.pages if p.PARENT == parent]
+        def pages():
+            for p in self.pages:
+                if p.PARENT == parent:
+                    yield p
+
```

Comment:
Why the inline function instead of the comprehension?

```python
pages = (p for p in self.pages if p.PARENT == parent)
``` 

Seems more elegant and clear to me.

Reply:
Well, that's a tuple vs a generator, so not totally the same thing.
But in this case we can prefer "elegant" I guess, done in [25964df](https://github.com/metabrainz/picard/pull/2436/commits/25964df99a633cd9601b59ce3b5cee3954040b1e)

Reply:
That's a [generator expression](https://peps.python.org/pep-0289/), so both are generators.

Reply:
Ah yes, my bad :)